### PR TITLE
Fat 2051

### PR DESCRIPTION
**Pull request by TetianaParanich**
Update already created auto tests for check summary table

### CODE REVIEW

**Review by MaksimTerekhov — COMMENTED** [review]
cypress/integration/data-import/mrc-import-file-for-update-items.spec.js

check imports in this file plesase (defined but never used)

File: cypress/integration/data-import/mrc-import-file-for-update-items.spec.js [hunk 1]
```diff
@@ -212,14 +213,80 @@ describe('ui-data-import: MARC file upload with the update of instance, holding,
...
+        name: mappingProfileNameForItem,
+        fillProfile: NewMappingProfile.fillItemMappingProfile },
+      actionProfile: { typeValue: NewActionProfile.folioRecordTypeValue.item,
+        name: actionProfileNameForItem,
+        action: 'Update (all record types except Orders)' }
+    }
+  ];
+
+  const collectionOfMatchProfiles = [
+    /*{
```
Comment (MaksimTerekhov) [inline cypress/integration/data-import/mrc-import-file-for-update-items.spec.js:244]: remove comments please

File: cypress/integration/data-import/mrc-import-file-for-update-items.spec.js [hunk 2]
```diff
@@ -212,14 +213,80 @@ describe('ui-data-import: MARC file upload with the update of instance, holding,
...
   beforeEach(() => {
-    cy.login(
-      Cypress.env('diku_login'),
-      Cypress.env('diku_password')
-    );
+    cy.login(Cypress.env('diku_login'), Cypress.env('diku_password'));
     cy.getAdminToken();
 
-    DataImport.checkUploadState();
+    /* DataImport.checkUploadState();
```
Comment (MaksimTerekhov) [inline cypress/integration/data-import/mrc-import-file-for-update-items.spec.js:289]: remove comments please

File: cypress/integration/data-import/mrc-import-file-for-update-items.spec.js [hunk 3]
```diff
@@ -240,24 +307,51 @@ describe('ui-data-import: MARC file upload with the update of instance, holding,
     DataImportSettingsJobProfiles.createJobProfileApi(testData.jobProfileForCreate)
       .then((bodyWithjobProfile) => {
         testData.jobProfileForCreate.id = bodyWithjobProfile.body.id;
-      });
+      }); */
```
Comment (MaksimTerekhov) [inline cypress/integration/data-import/mrc-import-file-for-update-items.spec.js:310]: comments

File: cypress/integration/data-import/mrc-import-file-for-update-items.spec.js [hunk 4]
```diff
@@ -240,24 +307,51 @@ describe('ui-data-import: MARC file upload with the update of instance, holding,
...
+    // delete downloads folder and created files in fixtures
+    FileManager.deleteFolder(Cypress.config('downloadsFolder'));
+    FileManager.deleteFile(`cypress/fixtures/${nameMarcFileForImportUpdate}`);
+    FileManager.deleteFile(`cypress/fixtures/${nameForCSVFile}`);
+  }); */
 
   it('C343335 MARC file upload with the update of instance, holding, and items (folijet)', { tags: [TestTypes.smoke, TestTypes.broken] }, () => {
     // upload a marc file for creating of the new instance, holding and item
-    cy.visit(TopMenu.dataImportPath);
+    /* cy.visit(TopMenu.dataImportPath);
```
Comment (MaksimTerekhov) [inline cypress/integration/data-import/mrc-import-file-for-update-items.spec.js:343]: comments

File: cypress/integration/data-import/mrc-import-file-for-update-items.spec.js [hunk 5]
```diff
@@ -240,24 +307,51 @@ describe('ui-data-import: MARC file upload with the update of instance, holding,
...
     DataImport.uploadFile('oneMarcBib.mrc', nameMarcFileForImportCreate);
     JobProfiles.searchJobProfileForImport(testData.jobProfileForCreate.profile.name);
     JobProfiles.runImportFile(nameMarcFileForImportCreate);
     Logs.openFileDetails(nameMarcFileForImportCreate);
     FileDetails.checkStatusInColumn(FileDetails.status.updated, FileDetails.columnName.srsMarc);
-    FileDetails.checkStatusInColumn(FileDetails.status.created, FileDetails.columnName.instance);
-    FileDetails.checkStatusInColumn(FileDetails.status.created, FileDetails.columnName.holdings);
-    FileDetails.checkStatusInColumn(FileDetails.status.created, FileDetails.columnName.item);
+    /*[FileDetails.columnName.instance,
```
Comment (MaksimTerekhov) [inline cypress/integration/data-import/mrc-import-file-for-update-items.spec.js:349]: comments

File: cypress/integration/data-import/mrc-import-file-for-update-items.spec.js [hunk 6]
```diff
@@ -288,90 +382,25 @@ describe('ui-data-import: MARC file upload with the update of instance, holding,
...
 
     // create Job profile
-    const jobProfileForUpdate = {
-      ...NewJobProfile.defaultJobProfile,
-      profileName: jobProfileNameUpdate,
-      acceptedType: NewJobProfile.acceptedDataType.marc
-    };
-
-    cy.visit(SettingsMenu.jobProfilePath);
+    /* cy.visit(SettingsMenu.jobProfilePath);
```
Comment (MaksimTerekhov) [inline cypress/integration/data-import/mrc-import-file-for-update-items.spec.js:403]: comments

File: cypress/support/fragments/data_import/logs/fileDetails.js [hunk 7]
```diff
@@ -29,7 +29,27 @@ const checkStatusInColumn = (specialStatus, specialColumnName) => {
...
+      .find(MultiColumnListRow({ indexRow: `row-${rowNumber}` }))
+      .find(MultiColumnListCell({ columnIndex: i, content: quantity }))
+      .exists());
+  }
+};
+
+const checkCreatedInvoiceISummaryTable = (quantity) => {
+  cy.expect(MultiColumnList({ id: 'job-summary-table' })
+    .find(MultiColumnListRow({ indexRow: 'row-0' }))
+    .find(MultiColumnListCell({ columnIndex: 1, content: quantity }))
```
Comment (MaksimTerekhov) [inline cypress/support/fragments/data_import/logs/fileDetails.js:44]: move to constant

File: cypress/integration/data-import/mrc-import-file-for-update-items.spec.js (hunk 1 above)
Comment (TetianaParanich) [reply cypress/integration/data-import/mrc-import-file-for-update-items.spec.js]: Done

File: cypress/integration/data-import/mrc-import-file-for-update-items.spec.js (hunk 2 above)
Comment (TetianaParanich) [reply cypress/integration/data-import/mrc-import-file-for-update-items.spec.js]: Done

File: cypress/integration/data-import/mrc-import-file-for-update-items.spec.js (hunk 3 above)
Comment (TetianaParanich) [reply cypress/integration/data-import/mrc-import-file-for-update-items.spec.js]: Done

File: cypress/integration/data-import/mrc-import-file-for-update-items.spec.js (hunk 4 above)
Comment (TetianaParanich) [reply cypress/integration/data-import/mrc-import-file-for-update-items.spec.js]: Done

File: cypress/integration/data-import/mrc-import-file-for-update-items.spec.js (hunk 5 above)
Comment (TetianaParanich) [reply cypress/integration/data-import/mrc-import-file-for-update-items.spec.js]: Done

File: cypress/integration/data-import/mrc-import-file-for-update-items.spec.js (hunk 6 above)
Comment (TetianaParanich) [reply cypress/integration/data-import/mrc-import-file-for-update-items.spec.js]: Done

File: cypress/support/fragments/data_import/logs/fileDetails.js (hunk 7 above)
Comment (TetianaParanich) [reply cypress/support/fragments/data_import/logs/fileDetails.js:46]: Done

File: cypress.json [hunk 8]
```diff
@@ -3,7 +3,7 @@
   "viewportWidth": 1920,
   "viewportHeight": 1080,
   "video": false,
-  "defaultCommandTimeout": 81000,
+  "defaultCommandTimeout": 101000,
```
Comment (MaksimTerekhov) [inline cypress.json:6]: +20 sec for waiting makes our test slower(

File: cypress/support/fragments/data_import/match_profiles/newMatchProfile.js [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+    cy.do(TextField({ name: 'profile.matchDetails[0].existingMatchExpression.fields[0].value' }).fillIn(field));
+  }
+  if (in1) {
+    cy.do(TextField({ name: 'profile.matchDetails[0].existingMatchExpression.fields[1].value' }).fillIn(in1));
+  }
+  if (in2) {
+    cy.do(TextField({ name: 'profile.matchDetails[0].existingMatchExpression.fields[2].value' }).fillIn(in2));
+  }
+  if (subfield) {
+    cy.do(TextField({ name: 'profile.matchDetails[0].existingMatchExpression.fields[3].value' }).fillIn(subfield));
```
Comment (MaksimTerekhov) [inline cypress/support/fragments/data_import/match_profiles/newMatchProfile.js:30]:
```
function fillExistingRecordFields(value, selector) => {
  const map = {
    field: 'profile.matchDetails[0].existingMatchExpression.fields[0].value',
    in1: 'profile.matchDetails[0].existingMatchExpression.fields[1].value',
    in2: 'profile.matchDetails[0].existingMatchExpression.fields[2].value',
    subfield: 'profile.matchDetails[0].existingMatchExpression.fields[3].value'
  }

  cy.do(TextField({ name: map[selector]).fillin(value)
```
You can call function like this fillExistingRecordFields('stringForFilling', 'in1')

Don't use in function more than one if please

File: cypress/support/fragments/data_import/match_profiles/newMatchProfile.js [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+  }
+  if (in1) {
+    cy.do(TextField({ name: 'profile.matchDetails[0].incomingMatchExpression.fields[1].value' }).fillIn(in1));
+  }
+  if (in2) {
+    cy.do(TextField({ name: 'profile.matchDetails[0].incomingMatchExpression.fields[2].value' }).fillIn(in2));
+  }
+  if (subfield) {
+    cy.do(TextField({ name: 'profile.matchDetails[0].incomingMatchExpression.fields[3].value' }).fillIn(subfield));
+  }
```
Comment (MaksimTerekhov) [inline cypress/support/fragments/data_import/match_profiles/newMatchProfile.js:46]: the same about if

**Review by MaksimTerekhov — COMMENTED** [review]
Also fix linter issue please

File: cypress.json (hunk 8 above)
Comment (TetianaParanich) [reply cypress.json:6]: yes but without this waiting data import tests are failed

File: cypress/support/fragments/data_import/match_profiles/newMatchProfile.js (hunk 9 above)
Comment (TetianaParanich) [reply cypress/support/fragments/data_import/match_profiles/newMatchProfile.js:30]: Done

File: cypress/support/fragments/data_import/match_profiles/newMatchProfile.js (hunk 10 above)
Comment (TetianaParanich) [reply cypress/support/fragments/data_import/match_profiles/newMatchProfile.js:46]: Done

**Review by MaksimTerekhov — APPROVED** [review]

File: cypress/integration/data-import/mrc-import-file-for-update-items.spec.js [hunk 11]
```diff
@@ -213,11 +214,77 @@ describe('ui-data-import: MARC file upload with the update of instance, holding,
...
+    profileName: jobProfileNameUpdate,
+    acceptedType: NewJobProfile.acceptedDataType.marc
+  };
+
   beforeEach(() => {
-    cy.login(
-      Cypress.env('diku_login'),
-      Cypress.env('diku_password')
-    );
+    cy.login(Cypress.env('diku_login'), Cypress.env('diku_password'));
```
Comment (penfriend) [inline cypress/integration/data-import/mrc-import-file-for-update-items.spec.js:287]: There is method cy.loginAsAdmin. Please use it

File: cypress/support/fragments/data-export/exportMappingProfile/modalSelectTransformations.js [hunk 12]
```diff
@@ -4,7 +4,7 @@ const ModalTransformation = Modal('Select transformations');
 
 export default {
   searchItemTransformationsByName: (name) => {
-    cy.get('div[class^=modal-] input[name=searchValue]').type(`${name}{enter}`);
+    cy.get('div[class^=modal-] input[name=searchValue]').clear().type(`${name}{enter}`);
```
Comment (penfriend) [inline cypress/support/fragments/data-export/exportMappingProfile/modalSelectTransformations.js:7]: A good practice is to create/extend the existed interactors to use it. Not to use cy.get directly. It is your decision to do it or not. It is just for information

File: cypress/support/fragments/data_import/match_profiles/newMatchProfile.js [hunk 13]
```diff
@@ -0,0 +1,89 @@
...
+  // select existing record type
+  if (existingRecordType === 'MARC_BIBLIOGRAPHIC') {
+    cy.do(Button({ dataId:'MARC_BIBLIOGRAPHIC' }).click());
+    fillIncomingRecordFields(incomingRecordFields.field, 'field');
+    cy.do(Select('Match criterion').choose(matchCriterion));
+    fillExistingRecordFields(existingRecordFields.field, 'field');
+  } else if (existingRecordType === 'INSTANCE') {
+    cy.intercept('/_/jsonSchemas?path=acq-models/mod-orders-storage/schemas/vendor_detail.json').as('getJson2');
+    cy.wait('@getJson2', getLongDelay());
+    cy.wait(1000);
```
Comment (penfriend) [inline cypress/support/fragments/data_import/match_profiles/newMatchProfile.js:62]: Please write a comment why you need this wait

File: cypress/support/fragments/data_import/searchInventory.js [hunk 14]
```diff
@@ -37,19 +37,17 @@ const getInstanceHRID = () => {
...
-  cy.get('#input-inventory-search-qindex').select('Instance HRID');
-  cy.do([
-    // Select({ id: 'input-inventory-search-qindex' }).choose('Instance HRID'),
-    TextField({ id: 'input-inventory-search' }).fillIn(id),
-    Button('Search').click()
-  ]);
+  InventoryInstances.waitContentLoading();
+  cy.do(Select({ id: 'input-inventory-search-qindex' }).choose('Instance HRID'));
+  cy.do(TextField({ id: 'input-inventory-search' }).fillIn(id));
+  cy.do(Button('Search').click());
```
Comment (penfriend) [inline cypress/support/fragments/data_import/searchInventory.js:50]: You can rewrite here:
cy.do(
Select({ id: 'input-inventory-search-qindex' }).choose('Instance HRID');
TextField({ id: 'input-inventory-search' }).fillIn(id)
Button('Search').click())

File: cypress/support/fragments/requests/requests.js [hunk 15]
```diff
@@ -293,6 +293,7 @@ export default {
   },
 
   verifyAssignedTags(tag) {
+    cy.wait(1500);
```
Comment (penfriend) [inline cypress/support/fragments/requests/requests.js:296]: write a comment why you need this wait

File: cypress/support/fragments/servicePoint/switchServicePoint.js [hunk 16]
```diff
@@ -23,6 +23,6 @@ export default {
       Button('Switch service point').click()
     ]);
     SelectServicePointModal.selectServicePoint(servicePoint);
-    cy.wait(5000);
+    cy.wait(2000);
```
Comment (penfriend) [inline cypress/support/fragments/servicePoint/switchServicePoint.js:26]: write a comment why you need this wait

File: cypress/integration/data-import/mrc-import-file-for-update-items.spec.js (hunk 11 above)
Comment (TetianaParanich) [reply cypress/integration/data-import/mrc-import-file-for-update-items.spec.js:287]: Done

File: cypress/support/fragments/data-export/exportMappingProfile/modalSelectTransformations.js (hunk 12 above)
Comment (TetianaParanich) [reply cypress/support/fragments/data-export/exportMappingProfile/modalSelectTransformations.js:7]: I got it

File: cypress/support/fragments/requests/requests.js (hunk 15 above)
Comment (TetianaParanich) [reply cypress/support/fragments/requests/requests.js:296]: Done

File: cypress/support/fragments/servicePoint/switchServicePoint.js (hunk 16 above)
Comment (TetianaParanich) [reply cypress/support/fragments/servicePoint/switchServicePoint.js:26]: Done

File: cypress/support/fragments/orders/newOrder.js [hunk 17]
```diff
@@ -22,6 +22,6 @@ export default {
   defaultOneTimeOrder: {
     id: uuid(),
     vendor: '',
-    orderType: 'One-Time'
+    orderType: 'One-time'
```
Comment (IhorBohdan) [inline cypress/support/fragments/orders/newOrder.js:25]: Are you sure this change is needed here?

**Review by IhorBohdan — APPROVED** [review]

File: cypress/support/fragments/orders/newOrder.js (hunk 17 above)
Comment (TetianaParanich) [reply cypress/support/fragments/orders/newOrder.js:25]: yep

**Review by penfriend — APPROVED** [review]

**Pull request merged**